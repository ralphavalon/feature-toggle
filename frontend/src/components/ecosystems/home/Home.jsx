import React, { useState } from 'react';
import { Row, Col, Button } from 'react-bootstrap';

import FeatureCard from '../../organisms/feature-card';
import FeatureList from '../../organisms/feature-list';

const Home = () => {

  const [isNewFeature, setIsNewFeature] = useState(false);

  return (
    <div>
      <Row>
        <Col className="col-sm-5 col-md-3">
          <Button variant="primary" type="button" onClick={() => setIsNewFeature(true)} >
            Add new feature
          </Button>
          {
            isNewFeature && <FeatureCard onSubmit={() => setIsNewFeature(false)} onCancel={() => setIsNewFeature(false)} />
          }
        </Col>
      </Row>
      <FeatureList />
    </div>
  );
};

export default Home;
