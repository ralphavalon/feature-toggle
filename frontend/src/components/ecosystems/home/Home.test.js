import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

describe('Home', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders feature card when clicking on Add new feature', () => {
    const wrapper = shallow(<Home />);
    wrapper.find('Button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders without feature card when clicking on Add new feature and then on Submit', () => {
    const wrapper = shallow(<Home />);
    wrapper.find('Button').simulate('click');
    wrapper.find('FeatureCardContainer').prop('onSubmit')();
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders without feature card when clicking on Add new feature and then on Cancel', () => {
    const wrapper = shallow(<Home />);
    wrapper.find('Button').simulate('click');
    wrapper.find('FeatureCardContainer').prop('onCancel')();
    expect(wrapper).toMatchSnapshot();
  });
})

