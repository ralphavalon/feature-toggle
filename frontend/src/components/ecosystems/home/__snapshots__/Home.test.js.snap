// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <Row
    noGutters={false}
  >
    <Col
      className="col-sm-5 col-md-3"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add new feature
      </Button>
    </Col>
  </Row>
  <FeatureListContainer />
</div>
`;

exports[`Home renders feature card when clicking on Add new feature 1`] = `
<div>
  <Row
    noGutters={false}
  >
    <Col
      className="col-sm-5 col-md-3"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add new feature
      </Button>
      <FeatureCardContainer
        onCancel={[Function]}
        onSubmit={[Function]}
      />
    </Col>
  </Row>
  <FeatureListContainer />
</div>
`;

exports[`Home renders without feature card when clicking on Add new feature and then on Cancel 1`] = `
<div>
  <Row
    noGutters={false}
  >
    <Col
      className="col-sm-5 col-md-3"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add new feature
      </Button>
    </Col>
  </Row>
  <FeatureListContainer />
</div>
`;

exports[`Home renders without feature card when clicking on Add new feature and then on Submit 1`] = `
<div>
  <Row
    noGutters={false}
  >
    <Col
      className="col-sm-5 col-md-3"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add new feature
      </Button>
    </Col>
  </Row>
  <FeatureListContainer />
</div>
`;
